### PR TITLE
Upgrade BCMDHD with cherry-picks from Angler's kernel

### PR DESCRIPTION
**Pull request by stellirin**
### loire
- [x] suzu
- [ ] kugo
### kitakmi
- [ ] satsuki
- [ ] satsuki_dsds
- [ ] sumire
- [ ] sumire_dsds
- [ ] suzuran
- [x] ivy
- [ ] ivy_dsds
- [x] karin
- [x] karin_windy
### kanuti
- [ ] tulip
### shinano
- [ ] leo
- [ ] aries
- [ ] sirius
- [ ] scorpion
- [ ] scorpion_windy
- [ ] castor
- [x] castor_windy
### rhine
- [ ] honami
- [ ] amami
- [ ] togari

### CODE REVIEW

**Comment by stellirin** [comment]
Needs to be tested on more devices, but on the devices I tried it's working

**Comment by humberos** [comment]
@AdFad666 Which device/platform did you try ?

**Comment by mcgi5sr2** [comment]
Tested this on Aries d5803, crashed out when searching for wifi network in one time installer app.

> http://pastebin.com/uER9WsWg

**Comment by stellirin** [comment]
@mcgi5sr2 @humberos I used git bisect to find and remove the offending commit. Could you please test again?

**Comment by humberos** [comment]
Good job!
Tested on Karin.

**Comment by rancidfrog** [comment]
Tested on aries using
`curl https://patch-diff.githubusercontent.com/raw/sonyxperiadev/kernel/pull/817.patch | git am`
I am able to connect to wifi.
Anything else I should test for?

**Comment by rcstar6696** [comment]
Tested on castor_windy and sirius both are ok

**Comment by humberos** [comment]
Also tested on M (including mimo patches).

**Comment by mcgi5sr2** [comment]
tested on Aries, seems ok

**Comment by mcgi5sr2** [comment]
latest patch set is causing a build error

`  CC      drivers/net/wireless/bcmdhd/wl_roam.o
  CC      net/netfilter/xt_connlimit.o
  CC      drivers/net/wireless/bcmdhd/wl_cfg80211.o
  CC      net/netfilter/xt_conntrack.o
/home/mcgi5sr2/AOSP/android/system/kernel/sony/msm/drivers/net/wireless/bcmdhd/wl_cfg80211.c: In function 'wl_cfg80211_disconnect':
/home/mcgi5sr2/AOSP/android/system/kernel/sony/msm/drivers/net/wireless/bcmdhd/wl_cfg80211.c:4186:13: warning: unused variable 'dhdp' [-Wunused-variable]
  dhd_pub_t *dhdp = (dhd_pub_t *)(cfg->pub);
             ^
/home/mcgi5sr2/AOSP/android/system/kernel/sony/msm/drivers/net/wireless/bcmdhd/wl_cfg80211.c: At top level:
/home/mcgi5sr2/AOSP/android/system/kernel/sony/msm/drivers/net/wireless/bcmdhd/wl_cfg80211.c:7835:2: warning: initialization from incompatible pointer type [enabled by default]
  .del_station = wl_cfg80211_del_station,
  ^
/home/mcgi5sr2/AOSP/android/system/kernel/sony/msm/drivers/net/wireless/bcmdhd/wl_cfg80211.c:7835:2: warning: (near initialization for 'wl_cfg80211_ops.del_station') [enabled by default]
/home/mcgi5sr2/AOSP/android/system/kernel/sony/msm/drivers/net/wireless/bcmdhd/wl_cfg80211.c: In function 'wl_inform_single_bss':
/home/mcgi5sr2/AOSP/android/system/kernel/sony/msm/drivers/net/wireless/bcmdhd/wl_cfg80211.c:8257:7: error: 'struct bcm_cfg80211' has no member named 'sched_scan_req'
   (cfg->sched_scan_req && !cfg->scan_request)) {
       ^
/home/mcgi5sr2/AOSP/android/system/kernel/sony/msm/drivers/net/wireless/bcmdhd/wl_cfg80211.c: In function 'wl_notify_escan_complete':
/home/mcgi5sr2/AOSP/android/system/kernel/sony/msm/drivers/net/wireless/bcmdhd/wl_cfg80211.c:10335:13: warning: unused variable 'dhdp' [-Wunused-variable]
  dhd_pub_t *dhdp = (dhd_pub_t *)(cfg->pub);
             ^
  CC      net/netfilter/xt_dscp.o
/home/mcgi5sr2/AOSP/android/system/kernel/sony/msm/scripts/Makefile.build:308: recipe for target 'drivers/net/wireless/bcmdhd/wl_cfg80211.o' failed
make[5]: *** [drivers/net/wireless/bcmdhd/wl_cfg80211.o] Error 1
/home/mcgi5sr2/AOSP/android/system/kernel/sony/msm/scripts/Makefile.build:455: recipe for target 'drivers/net/wireless/bcmdhd' failed
make[4]: *** [drivers/net/wireless/bcmdhd] Error 2
/home/mcgi5sr2/AOSP/android/system/kernel/sony/msm/scripts/Makefile.build:455: recipe for target 'drivers/net/wireless' failed
make[3]: *** [drivers/net/wireless] Error 2
/home/mcgi5sr2/AOSP/android/system/kernel/sony/msm/scripts/Makefile.build:455: recipe for target 'drivers/net' failed
make[2]: *** [drivers/net] Error 2
/home/mcgi5sr2/AOSP/android/system/kernel/sony/msm/Makefile:821: recipe for target 'drivers' failed
make[1]: *** [drivers] Error 2
make[1]: *** Waiting for unfinished jobs....
`

File: drivers/net/wireless/bcmdhd/wl_cfg80211.c [hunk 1]
```diff
@@ -8167,6 +8252,50 @@ static s32 wl_inform_single_bss(struct bcm_cfg80211 *cfg, struct wl_bss_info *bi
 #else
 	cfg80211_put_bss(cbss);
 #endif /* (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 9, 0) */
+
+	if (DBG_RING_ACTIVE(dhdp, DHD_EVENT_RING_ID) &&
+		(cfg->sched_scan_req && !cfg->scan_request)) {
```
Comment (mcgi5sr2) [inline drivers/net/wireless/bcmdhd/wl_cfg80211.c:8257]: /kernel/sony/msm/drivers/net/wireless/bcmdhd/wl_cfg80211.c:8257:7: error: 'struct bcm_cfg80211' has no member named 'sched_scan_req'
   (cfg->sched_scan_req && !cfg->scan_request)) {
       ^

Comment (mcgi5sr2) [inline drivers/net/wireless/bcmdhd/wl_cfg80211.c]: causing current break in build process

**Review by mcgi5sr2 — COMMENTED** [review]
error in build process

**Review by mcgi5sr2 — COMMENTED** [review]
commit causing break in build process
    net: wireless: bcmdhd: logging of PNO activities  …           cc14f00

File: drivers/net/wireless/bcmdhd/wl_cfg80211.c (hunk 1 above)
Comment (mcgi5sr2) [inline drivers/net/wireless/bcmdhd/wl_cfg80211.c]: net: wireless: bcmdhd: logging of PNO activities  …			cc14f00

**Comment by pablomh** [comment]
@AdFad666, I think you can skip kanuti (tulip) from the list and add tone (dora / kagura):

https://github.com/search?utf8=%E2%9C%93&q=%22BOARD_WLAN_DEVICE+++++++++++%3A%3D+bcmdhd%22+user%3Asonyxperiadev&type=Code&ref=searchresults

**Pull request closed without merging**